L46-core-tls-credential-API.md

```diff
@@ -0,0 +1,472 @@
+  // extension in the peer's leaf certificate.
+  absl::Span<absl::string_view> uri_names() const;
+
+  // The (possibly empty) list of DNS names from the Subject Alternative Name
+  // extension in the peer's leaf certificate.
+  absl::Span<absl::string_view> dns_names() const;
+
+  // The (possibly empty) list of email names from the Subject Alternative Name
+  // extension in the peer's leaf certificate.
+  absl::Span<absl::string_view> email_names() const;
+
+  // The (possibly empty) list of IP names from the Subject Alternative Name
+  // extension in the peer's leaf certificate.
+  absl::Span<absl::string_view> ip_names() const;
+}
+
+// The base class of all verifier implementations. Note that custom
+// verifier implementations can compose their functionality with existing
+// implementations of this interface, such as HostNameVerifier, by delegating
+// to an instance of that class.
+class CertificateVerifierInterface {
+ public:
+  virtual ~CertificateVerifierInterface() = default;
+
+  // Performs a custom verification check.
+  // Verification occurs asynchronously. gRPC calls `Verify` on this interface.
+  // The callback must be called when verification is complete and the
+  // verification result must be populated in the output parameter of the
+  // callback function.
+  //
+  // request: the verification information associated with this request
+  // callback: The function that gRPC will invoke when the verifier has already
+  //           completed its asynchronous check. Callers can use this function
+  //           to perform any additional checks. The input parameter of the
+  //           any::Invocable indicates the status of the verifier check.
+  virtual void Verify(std::shared_ptr<VerifiedPeerCertificateChainInfo> verified_chain_info,
+              absl::AnyInvocable<void(absl::Status)> callback) = 0;
+
+  // Cancels a verification request previously started via Verify().
+  // Used when the connection attempt times out or is cancelled while an async
+  // verification request is pending.
+  //
+  // request: the verification information associated with this request
+  virtual void Cancel(std::shared_ptr<VerifiedPeerCertificateChainInfo> verified_chain_info) = 0;
+};
+
+// A CertificateVerifier that will perform hostname verification, to see if the
+// target name set from the client side matches the identity information
+// specified on the server's certificate.
+// This will be used as the default verifier if no verification is otherwise
+// set.
+class HostNameCertificateVerifier : public CertificateVerifierInterface {
+ public:
+  HostNameCertificateVerifier();
+};
+```
+
+#### Custom Chain Building
+```c++
+
+
+class UnverifiedPeerChain {
+  std::vector<absl::string_view> peer_cert_chain_der;
+}
+
+
+class CustomChainBuilderInterface {
+public:
+ virtual ~CustomChainBuilderInterface() = default;
+
+ // Performs custom chain building. gRPC invokes this function. When Chain
```

Comment:
nit: Consider omitting "gRPC invokes this function.", as I don't think it adds anything.

Reply:
Removed